#! /bin/sh /usr/share/dpatch/dpatch-run
## alsa-patch.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Makes ALSA the default handler for audio.

@DPATCH@
diff -urNad mythtv-0.21.0~fixes15967~/programs/mythfrontend/globalsettings.cpp mythtv-0.21.0~fixes15967/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.21.0~fixes15967~/programs/mythfrontend/globalsettings.cpp	2008-02-12 11:21:57.000000000 -0600
+++ mythtv-0.21.0~fixes15967/programs/mythfrontend/globalsettings.cpp	2008-02-12 21:25:07.000000000 -0600
@@ -41,6 +41,15 @@
     HostComboBox *gc = new HostComboBox("AudioOutputDevice", true);
     gc->setLabel(QObject::tr("Audio output device"));
 
+#ifdef USING_ALSA
+    gc->addSelection("ALSA:default",       "ALSA:default");
+    gc->addSelection("ALSA:spdif",         "ALSA:spdif");
+    gc->addSelection("ALSA:surround51",    "ALSA:surround51");
+    gc->addSelection("ALSA:analog",        "ALSA:analog");
+    gc->addSelection("ALSA:digital",       "ALSA:digital");
+    gc->addSelection("ALSA:mixed-analog",  "ALSA:mixed-analog");
+    gc->addSelection("ALSA:mixed-digital", "ALSA:mixed-digital");
+#endif
 #ifdef USING_OSS
     QDir dev("/dev", "dsp*", QDir::Name, QDir::System);
     gc->fillSelectionsFromDir(dev);
@@ -56,15 +65,6 @@
         gc->fillSelectionsFromDir(dev);
     }
 #endif
-#ifdef USING_ALSA
-    gc->addSelection("ALSA:default",       "ALSA:default");
-    gc->addSelection("ALSA:spdif",         "ALSA:spdif");
-    gc->addSelection("ALSA:surround51",    "ALSA:surround51");
-    gc->addSelection("ALSA:analog",        "ALSA:analog");
-    gc->addSelection("ALSA:digital",       "ALSA:digital");
-    gc->addSelection("ALSA:mixed-analog",  "ALSA:mixed-analog");
-    gc->addSelection("ALSA:mixed-digital", "ALSA:mixed-digital");
-#endif
 #ifdef USING_ARTS
     gc->addSelection("ARTS:", "ARTS:");
 #endif
